src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheck.java

```diff
@@ -608,22 +627,67 @@ else if (assignDepth >= 1) {
      */
     private static boolean unnecessaryParenAroundOperators(DetailAST ast) {
         final int type = ast.getType();
-        final int parentType = ast.getParent().getType();
         final boolean isConditional = TokenUtil.isOfType(type, CONDITIONALS_AND_RELATIONAL);
-        boolean result = TokenUtil.isOfType(parentType, CONDITIONALS_AND_RELATIONAL);
+        final boolean isBitwise = TokenUtil.isOfType(type, BITWISE_BINARY_OPERATORS);
+        final boolean result;
         if (isConditional) {
-            if (type == TokenTypes.LOR) {
-                result = result && !TokenUtil.isOfType(parentType, TokenTypes.LAND);
-            }
-            result = result && !TokenUtil.isOfType(parentType, TokenTypes.EQUAL,
-                TokenTypes.NOT_EQUAL);
+            result = checkConditionalOperator(ast);
+        }
+        else if (isBitwise) {
+            result = checkBitwiseBinaryOperator(ast);
         }
         else {
-            result = result && TokenUtil.isOfType(type, UNARY_AND_POSTFIX);
+            result = (TokenUtil.isOfType(ast.getParent().getType(), CONDITIONALS_AND_RELATIONAL)
+                    || TokenUtil.isOfType(ast.getParent().getType(), BITWISE_BINARY_OPERATORS))
+                    && TokenUtil.isOfType(type, UNARY_AND_POSTFIX);
         }
         return result;
     }
 
+    /**
+     * Check if conditional or logical operator has unnecessary parentheses.
+     *
+     * @param ast to check if surrounded by unnecessary parentheses
+     * @return true if Unnecessary parenthesis
+     */
+    private static boolean checkConditionalOperator(DetailAST ast) {
+        final int type = ast.getType();
+        final int parentType = ast.getParent().getType();
+        boolean result = true;
+        if (type == TokenTypes.LOR) {
+            result = !TokenUtil.isOfType(parentType, TokenTypes.LAND)
+                    && !TokenUtil.isOfType(parentType, BITWISE_BINARY_OPERATORS);
+        }
+        else if (type == TokenTypes.LAND) {
+            result = !TokenUtil.isOfType(parentType, BITWISE_BINARY_OPERATORS);
+        }
+        return result && !TokenUtil.isOfType(parentType, TokenTypes.EQUAL, TokenTypes.NOT_EQUAL)
+                && (TokenUtil.isOfType(parentType, CONDITIONALS_AND_RELATIONAL)
+                || TokenUtil.isOfType(parentType, BITWISE_BINARY_OPERATORS));
```

Comment:
Please extract this to helper method, let's call it 'isBitWiseBinaryOrConditionalOperator`, we can use it in three places.

Reply:
done